src/infrastructure/controllers/store/root-store/navigation-store.tsx

```diff
@@ -0,0 +1,32 @@
+import { action, makeAutoObservable, observable } from 'mobx';
+import { PagesEnum } from '~/domain/interfaces/enum/pages-enum';
+
+class NavigationStore {
+    navigate: any = () => {};
+    activeScreen: PagesEnum;
+
+    constructor() {
+        this.activeScreen = PagesEnum.HomePage;
+
+        makeAutoObservable(
+            this,
+            {
+                navigate: observable,
+                activeScreen: observable,
+                setNavigate: action,
+                setActiveScreen: action
+            },
+            { autoBind: true }
+        );
+    }
+
+    setNavigate = (navigate: any) => {
```

Comment:
Du coup on pourrait remplacer le any ici aussi
```suggestion
    setNavigate = (navigate: (route: string) => void) => {
```